segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/noop/NoOpStorageSystemDataWriteReadTests.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.noop;
+
+import io.pravega.segmentstore.storage.AsyncStorageWrapper;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageTestBase;
+import io.pravega.segmentstore.storage.SyncStorage;
+import io.pravega.segmentstore.storage.mocks.InMemoryStorageFactory;
+import org.junit.Before;
+
+/**
+ * This is to test Storage in No-Op mode for system segments only.
+ *
+ * For system segment storage, operations on both directions (write and read) should just work as before.
+ * In other words, No-Op layer is totally transparent for system segments.
+ */
+public class NoOpStorageSystemDataWriteReadTests extends StorageTestBase {
+
+    private SyncStorage systemStorage;
+    private StorageExtraConfig config;
+
+    @Before
+    public void setUp() {
+        //In this test only system segments are being tested.
+        isTestingSystemSegment = true;
+        systemStorage = new InMemoryStorageFactory(executorService()).createSyncStorage();
+        config = StorageExtraConfig.builder().with(StorageExtraConfig.STORAGE_NO_OP_MODE, true).build();
+    }
+
+    @Override
+    protected Storage createStorage() {
+        return new AsyncStorageWrapper(new NoOpStorage(config, systemStorage, null), executorService());
+    }
+
+    @Override
+    public void testFencing() {
+    }
```

Comment:
Did you purposefully leave this empty? If so, you should write "This method intentionally left blank", otherwise someone might think this is incomplete.

Reply:
Fixed. The method must show up as it's defined as abstract method in super